### PR TITLE
Unicycler: Add option to use only long reads

### PR DESCRIPTION
**Pull request by willemdek11**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [X] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

**Comment by bgruening** [comment]
@willemdek11 we just deployed a new unicylcer conda package and I adopted this PR. The explicit racon dependency is not needed anymore.

From the relase logs it seems there are a few more parameters that can be added if you like:

```
    New option --min_anchor_seg_len to explicitly set the minimum length for scaffolding anchors
    New option --kmers to explicitly set SPAdes assembly k-mer sizes
    New option --spades_tmp_dir to specify a location for SPAdes' temporary directory (solves #69 and #82, thanks @rec3141!)
```

But its not needed to get this one merged I guess. Your call. Thanks for your work!

**Comment by willemdek11** [comment]
Okay thanks @bgruening, this is enough for the moment.

**Comment by willemdek11** [comment]
@bgruening Alright, sorry misunderstood your comment. I reopened the pull request. The other options are not needed for me at least.

**Comment by bgruening** [comment]
Thanks @willemdek11! Looks great to me!

**Comment by shiltemann** [comment]
@willemdek11 thanks a lot! ..could you make or update a test that sets a value for these new parameters? (locally I am getting an `--min_anchor_seg_len: unrecognized parameter` error, so wondering if same happens in travis or if it's just me)

**Comment by shiltemann** [comment]
@willemdek11 ok, looks like it was just me, thanks for adding it to the tests, looks good to me too! :+1:

File: tools/unicycler/unicycler.xml [hunk 1]
```diff
@@ -135,10 +145,12 @@ $pilon.no_pilon
...
+        <param argument="--min_anchor_seg_len" type="integer" optional="true" label="Unicycler will not use segments shorter than this as scaffolding anchors"/>
         <section name="spades" expanded="False" title="SPAdes options" help="Unicycler uses SPAdes to construct assembly graphs. You can modify some of the SPAdes settings here. Use this ONLY if you know what you are doing!">
             <param argument="--no_correct" type="boolean" checked="false" truevalue="--no_correct" falsevalue="" label="Skip SPAdes error correction step" help="This option turns off SPAdes error correction. Generally it is highly recommended to use correction."/>
             <param argument="--min_kmer_frac" type="float" min="0" max="1" value="0.2" label="Lowest k-mer size for SPAdes assembly, expressed as a fraction of the read length"/>
             <param argument="--max_kmer_frac" type="float" min="0" max="1" value="0.95" label="Highest k-mer size for SPAdes assembly, expressed as a fraction of the read length"/>
+            <param argument="--kmers" type="text" value="" optional="true" label="Exact k-mers to use for SPAdes assembly, comma-separated"/>
```
Comment (shiltemann) [inline tools/unicycler/unicycler.xml]: You could consider adding a `<validator>` tag here with a regex to ensure users enter a correctly formatted string (e.g. only digits and commas)

[example](https://github.com/galaxyproject/tools-iuc/blob/57493fdc88e6128bed127993464f2a6480e2774d/tools/bcftools/bcftools_isec.xml#L56)

File: tools/unicycler/unicycler.xml [hunk 2]
```diff
@@ -135,10 +145,12 @@ $pilon.no_pilon
         </param>
         <param argument="--min_fasta_length" type="integer" value="100" label="Exclude contigs from the FASTA file which are shorter than this length (bp)"/>
         <param argument="--linear_seqs" type="integer" value="0" label="The expected number of linear (i.e. non-circular) sequences in the assembly"/>
+        <param argument="--min_anchor_seg_len" type="integer" optional="true" label="Unicycler will not use segments shorter than this as scaffolding anchors"/>
```
Comment (shiltemann) [inline tools/unicycler/unicycler.xml]: add `min="0"`?

**Comment by shiltemann** [comment]
thanks @willemdek11!

**Comment by bgruening** [comment]
Cool, @willemdek11 very good work. Thanks a lot!

**Pull request merged**